MatrixSDK/Utils/MXLog.swift

```diff
@@ -194,3 +194,40 @@ private var logger: SwiftyBeaver.Type = {
         return "\(message) - \(details)"
     }
 }
+
+/// Convenience wrapper around `MXLog` which formats all logs as "[Name] function: <message>"
+struct MXNamedLog {
```

Comment:
Interesting! If we want global usage of same log naming we can also make a `MXLogModule` like:
```
public enum LogModule: String {
  case webService
  case parsing
  case database
  case geolocation
}

extension LogModule {
  
  // MARK: Public

    func debug(_ message: String, function: String = #function) {
         MXLog.debug(formattedMessage(message, function: function))
     }
  
  // MARK: Private
    private func formattedMessage(_ message: String, function: String) -> String {
         return "[\(self.rawValue))] \(function): \(message)"
     }
}

```
And then use it like: `MXLogModule.geolocation.debug("Test")`

Reply:
Yea that sounds good, thought perhaps yet another use case, as it assumes a smaller number of stable areas, as your examples above. In my case the context equals any given class, which would be too many areas to keep a track of in an enum. I think both these use cases are complimentary